electron-app/main/menu.ts

```diff
@@ -85,4 +86,99 @@ if (process.platform === 'darwin') {
   MENU.push(HELP_MENU);
 }
 
-export default MENU;
+export function popupContextMenu(
+  window: BrowserWindow,
+  isDevelopment: boolean,
+  props: Electron.ContextMenuParams
+) {
+  const editFlags: any = props.editFlags;
+  const hasText = props.selectionText.trim().length > 0;
+  const roleEnabled = (role: string) => editFlags[`can${capitalize(role)}`] && hasText;
+
+  // Not using the `enable` nor `visible` properties of MenuConstructorOptions
+  // As they do not appear to work w/ Electron 1.8.4 on Mac OS X
+  let ctxMenuTmpl: MenuItemConstructorOptions[] = [];
+
+  if (roleEnabled('cut') && props.isEditable) {
+    ctxMenuTmpl.push({
+      label: 'Cut',
+      accelerator: 'CmdOrCtrl+X',
+      role: 'cut'
+    });
+  }
+
+  if (roleEnabled('copy') && hasText) {
+    ctxMenuTmpl.push({
+      label: 'Copy',
+      accelerator: 'CmdOrCtrl+C',
+      role: 'copy'
+    });
+  }
+
+  // For some reason, editFlags.canCopy is often false but the keyboard shortcut to paste still works
+  // let's only check if the currently focused element is editable
+  if (props.isEditable) {
+    ctxMenuTmpl.push({
+      label: 'Paste',
+      accelerator: 'CmdOrCtrl+V',
+      role: 'paste'
+    });
+  }
+
+  // skipping adding "Save Image" as it requires adding the electron-dl dependency
+  /* 
+    if (props.mediaType == 'image') {
+      ctxMenuTmpl = [{
+        type: 'separator'
+      }, {
+        id: 'save',
+        label: 'Save Image',
+        click() {
+          download(window, props.srcURL);
+        }
+      }];
+    } */
+
+  if (props.linkURL && props.mediaType === 'none') {
+    ctxMenuTmpl = ctxMenuTmpl.concat([
+      { type: 'separator' },
+      {
+        id: 'copyLink',
+        label: 'Copy Link',
+        click() {
+          if (process.platform === 'darwin') {
+            clipboard.writeBookmark(props.linkText, props.linkURL);
+          } else {
+            clipboard.writeText(props.linkURL);
+          }
+        }
+      },
+      {
+        id: 'openLink',
+        label: 'Open Link',
+        click() {
+          shell.openExternal(props.linkURL);
```

Comment:
Make sure this is an `https` URL before running openExternal, this could execute code.

Reply:
You can see an example of this here: https://github.com/MyCryptoHQ/MyCrypto/blob/develop/electron-app/main/window.ts#L38 perhaps it could be broken out into a util function.